### PR TITLE
Docker: build frontend dependencies with node and yarn

### PR DESCRIPTION
**Pull request by ArthurHoaro**
Based on #1072 - please only check the latest commit.

Tested locally, it seems to work fine.
Currently, it can be tested by replacing the code retrieval with:
```
+RUN curl -L https://github.com/ArthurHoaro/Shaarli/archive/feature/modern-front-end.tar.gz | tar xzf - \
+    && mv Shaarli-feature-modern-front-end shaarli \
```

### CODE REVIEW

File: docker/alpine/Dockerfile.master [hunk 1]
```diff
@@ -8,6 +8,16 @@ RUN curl -L https://github.com/shaarli/Shaarli/archive/master.tar.gz | tar xzf -
     && composer --prefer-dist --no-dev install
 
 # Stage 2:
+# - Frontend dependencies
+FROM node:latest as node
+RUN echo `pwd`
```
Comment (virtualtam) [inline docker/alpine/Dockerfile.master]: debug?

File: docker/alpine/Dockerfile.master [hunk 2]
```diff
@@ -8,6 +8,16 @@ RUN curl -L https://github.com/shaarli/Shaarli/archive/master.tar.gz | tar xzf -
     && composer --prefer-dist --no-dev install
 
 # Stage 2:
+# - Frontend dependencies
+FROM node:latest as node
```
Comment (virtualtam) [inline docker/alpine/Dockerfile.master]: suggestions:
- use an Alpine-based image for size & speed
- pin the major (and maybe minor?) NodeJS version

e.g. `FROM node:9.9-alpine`

Comment (ArthurHoaro) [reply docker/alpine/Dockerfile.master]: Hm, good idea, the size ratio is 1/10:

```
node                 9.9-alpine            3e60aa6db49b        5 days ago           68.4MB
node                 latest              4885ab8871c2        5 days ago           673MB
```

File: docker/alpine/Dockerfile.master (hunk 1 above)
Comment (ArthurHoaro) [reply docker/alpine/Dockerfile.master]: :speak_no_evil:

**Pull request merged**